lib/domain.js

```diff
@@ -399,32 +399,32 @@ EventEmitter.init = function() {
 const eventEmit = EventEmitter.prototype.emit;
 EventEmitter.prototype.emit = function emit(...args) {
   const domain = this.domain;
-  if (domain === null || domain === undefined || this === process) {
-    return Reflect.apply(eventEmit, this, args);
-  }
 
   const type = args[0];
-  // edge case: if there is a domain and an existing non error object is given,
-  // it should not be errorized
-  // see test/parallel/test-event-emitter-no-error-provided-to-error-event.js
-  if (type === 'error' && args.length > 1 && args[1] &&
-    !(args[1] instanceof Error)) {
-    domain.emit('error', args[1]);
-    return false;
-  }
+  const doError = type === 'error' &&
```

Comment:
This `doError` is opposite of the `doError` in the old `events.js` implementation. Logically it is not wrong but it makes it hard to compare the implementations.

Reply:
That seems more like a naming issue. Maybe just name it `shouldEmitError` instead of `doError`?

Reply:
I'm fine with a renaming. But you should make it clear if the emit is on the original `EventEmitter` or the `domain` instance.